main.tf

```diff
@@ -41,6 +41,18 @@ resource "google_compute_subnetwork" "subnetwork" {
   secondary_ip_range = "${var.secondary_ranges[lookup(var.subnets[count.index], "subnet_name")]}"
 }
 
+resource "google_compute_subnetwork" "no_secondary_subnetwork" {
```

Comment:
Is it possible to collapse these into a single resource? Having separate resources has major implications for state as it means adding a secondary range to a subnet would be impossible (it would attempt to delete the subnet, then create it fresh).

Reply:
> Adding secondary networks post creation from my dealings has not been a common occurrence and is only used within private GKE clusters in shared-vpc. 

It's not especially uncommon though, as you will occasionally need to start using GKE on an existing network.

> Currently the way the module is constructed it's forcing to burn quite a bit of your supernet scopes. So the likely transition for this would be to create the new segment in the network and drain your resources over to the new networks.

That's still pretty suboptimal. Often enterprises have constrained subnet space so needing to create a whole new primary subnet to introduce secondary ranges isn't ideal. Could you do a bit more discovery here? I'm wondering if it's possible to have `secondary_ip_range` default to an empty list in cases where it's not needed.

Reply:
> Could you do a bit more discovery here? I'm wondering if it's possible to have secondary_ip_range default to an empty list in cases where it's not needed.

Yeah I can look into that further